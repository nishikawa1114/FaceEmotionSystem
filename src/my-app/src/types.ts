export interface ImageUrl {
    id: number;
    url: string;
}

export interface Emotion {
    anger: number;
    contempt: number;
    disgust: number;
    fear: number;
    happiness: number;
    neutral: number;
    sadness: number;
    surprise: number;
}

export interface FaceAttributes {
    emotion: Emotion;
}

export interface FaceRectangle {
    top: number;
    left: number;
    width: number;
    height: number;
}

// 分析結果のデータ型
export interface ResultData {
    faceID: string;
    faceRectangle: FaceRectangle;
    faceAttributes: FaceAttributes;
}

export interface MeanFaceAttributes {
    faceAttributes: FaceAttributes;
}

export interface ResponseData {
    total: number;
    mean: MeanFaceAttributes
    resultData: Array<ResultData>;
}

// エラーID
export enum ErrorId {
    NOT_ERROR,
    ERROR_IMAGE_NOT_EXIST,
    ERROR_ANALYZE_RESULT_EMPTY
}