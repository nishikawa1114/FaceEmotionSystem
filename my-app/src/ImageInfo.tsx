import React from 'react';
import { getSupportedCodeFixes } from 'typescript';
import './index.css';

interface Image {
    id: number;
    url: string;
    // name: string;
    // date: string;
}


interface ImageInfoState {
    id: number
    name: string
    date: string
    url: string
}

interface ImageInfoProps {
    image: Image
}

export class ImageInfo extends React.Component<ImageInfoProps, ImageInfoState> {
    root: HTMLDivElement | null | undefined;

    public constructor(props: ImageInfoProps) {
        super(props);
        this.state = {
            id: props.image.id,
            name: this.getName(props.image.url),
            date: this.getDate(props.image.url),
            url: this.props.image.url,
        }
    }

    getDate(str: string) {
        // let strDate: string = String(str.match(/(\d{4}\/\d{2}\/\d{2})/));
        let strDate = String(str.match(/\d{4}\/\d{2}\/\d{2}/));
        return strDate;
    }

    getName(str: string) {
        let temp: string = String(str.split('images/').pop());
        let name: string = String(temp.split('/').shift());
        return name;
    }

    render() {
        return (
            <div className="image_info">
                {/* 画像 */}
                <div ref={ref => this.root = ref}>
                    <img 
                        src={this.props.image.url}
                    />
                    <br />
                    <form>
                        <input type="checkbox" value={this.props.image.id} />
                    </form>
                {/* ユーザー名,日付 */}
                    {this.props.image.id}<br/>
                    {this.state.name}<br/>
                    {this.state.date}
                </div>
            </div>
        )
    }
}


// class ImageIngo {
//     private _id: number
//     private _name: string
//     private _date: Date
//     private _url: string

//     constructor(id: number, name: string, date: Date, url: string) {
//         this._id = id
//         this._name = name
//         this._date = date
//         this._url = url
//     }

//     get id() {
//         return this._id
//     }

//     get name() {
//         return this._name
//     }

//     get date() {
//         return this._date
//     }

//     get url() {
//         return this._url
//     }
// }