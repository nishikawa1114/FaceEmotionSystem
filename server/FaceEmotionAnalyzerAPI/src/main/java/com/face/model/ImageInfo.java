package com.face.model;

public class ImageInfo {
	public String url;
	
	public void setUrl(String url) {
		this.url = url;
	}

	public String getUrl() {
		return url;
	}
}
