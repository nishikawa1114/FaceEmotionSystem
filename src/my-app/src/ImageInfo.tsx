import React from 'react';
import './index.css';
import { Image } from './types';

interface ImageInfoState {
    id: number
    name: string
    date: string
    url: string
}

interface ImageInfoProps {
    image: Image;
    checked: boolean;
    onClick: () => void;
}

export class ImageInfo extends React.Component<ImageInfoProps, ImageInfoState> {
    public constructor(props: ImageInfoProps) {
        super(props);
        this.state = {
            id: props.image.id,
            name: this.getName(props.image.url),
            date: this.getDate(props.image.url),
            url: this.props.image.url,
        }
    }

    // URLから日付を返す
    getDate(str: string) {
        const strDate = String(str.match(/\d{4}\/\d{2}\/\d{2}/));
        return strDate;
    }

    // URLからユーザー名を返す
    getName(str: string) {
        const temp: string = String(str.split('images/').pop());
        const name: string = String(temp.split('/').shift());
        return name;
    }

    render() {
        return (
            <div className="image_info">
                {/* 画像 */}
                <img
                    src={this.props.image.url}
                    className="image"
                    onClick={this.props.onClick}
                    alt={this.props.image.url}
                />
                <form>
                    <input
                        type="checkbox"
                        className="image_checkbox"
                        value={this.props.image.url}
                        checked={this.props.checked}
                    />
                </form>
                {/* ユーザー名,日付 */}
                {this.state.name}<br />
                {this.state.date}
            </div>
        )
    }
}