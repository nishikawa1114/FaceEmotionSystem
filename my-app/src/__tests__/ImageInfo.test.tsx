import { ImageInfo } from "../ImageInfo";
import { create } from "react-test-renderer";
import React from "react";
import { Image } from "../types";


const img: Image = {id: 1, url: "https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"};

it("unchecked", () => {
    const tree = create(
        <ImageInfo
            image={img}
            onClick={() => { }}
            checked={false}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();

})

it("checked", () => {
    const tree = create(
        <ImageInfo
            image={img}
            onClick={() => { }}
            checked={true}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();

})