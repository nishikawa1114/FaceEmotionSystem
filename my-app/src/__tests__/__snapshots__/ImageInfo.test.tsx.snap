// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checked 1`] = `
<div
  className="image_info"
>
  <img
    alt="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
    className="image"
    onClick={[Function]}
    src="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
  />
  <form>
    <input
      checked={true}
      className="image_checkbox"
      type="checkbox"
      value="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
    />
  </form>
  steve
  <br />
  2020/10/15
</div>
`;

exports[`unchecked 1`] = `
<div
  className="image_info"
>
  <img
    alt="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
    className="image"
    onClick={[Function]}
    src="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
  />
  <form>
    <input
      checked={false}
      className="image_checkbox"
      type="checkbox"
      value="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
    />
  </form>
  steve
  <br />
  2020/10/15
</div>
`;
